### ISSUE TITLE
Class overrides a field of a parent class - more specific type hints false positive

### ISSUE THREAD

**Issue by rbairwell** [kalessil/phpinspectionsea#986]
PHPInspections currently reports 'Class overrides a field of a parent class' in class B of the following:
```php
class A {
    /**
     * @var \DOMNode $domNode
     */
    protected $domNode;

   ...
}
class B extends A {
    /**
     * @var \DOMText $domNode
     */
    protected $domNode;

  ...
}
```

And, yes, as far as the PHP code is concerned, they are identical - but to PHPStorm, the latter provides better type hinting for that class. Would it be possible to suppress the notice if the hinted type extends the former type?

**Comment by kalessil** [comment]
Hi @rbairwell, understood.

Will this be a suitable solution for you (does the type-hinting override):
```php
/** @property  \DOMText $domNode */
class B extends A {
  ...
}

Or we still need to consider the case as a false-positive?
```

**Comment by kalessil** [comment]
ping @rbairwell

**Comment by kalessil** [comment]
No worries =) i updated our documentation as a follow up, as this a common case which is not visible from docs.
